server/neptune/workflows/internal/activities/terraform/summary.go

```diff
@@ -0,0 +1,61 @@
+package terraform
+
+import (
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-json"
+	"github.com/pkg/errors"
+)
+
+type ResourceSummary struct {
+	Address string
+}
+
+type PlanSummary struct {
+	Creations []ResourceSummary
+	Deletions []ResourceSummary
+	Updates   []ResourceSummary
+}
+
+// Generates a super simple plan summary with changes grouped by action
+// creation, deletion, update.
+// changes are only represented using addresses for now.
+func NewPlanSummaryFromJSON(b []byte) (PlanSummary, error) {
+	if len(b) == 0 {
+		return PlanSummary{}, nil
+	}
+	var plan tfjson.Plan
+	err := json.Unmarshal(b, &plan)
+
+	if err != nil {
+		return PlanSummary{}, errors.Wrap(err, "parsing plan json")
+	}
+
+	var creations []ResourceSummary
+	var deletions []ResourceSummary
+	var updates []ResourceSummary
+	for _, c := range plan.ResourceChanges {
+		summary := ResourceSummary{
+			Address: c.Address,
+		}
+		actions := c.Change.Actions
+		if actions.Delete() || actions.Replace() || actions.DestroyBeforeCreate() {
+			deletions = append(deletions, summary)
+		}
+
+		if actions.Create() || actions.CreateBeforeDestroy() || actions.Replace() {
```

Comment:
I think this can be simplified to: `if actions.Delete() || actions.Replace() {}`
and `if actions.Create() || actions.Replace() {}`. 

Looks like the `Replace` function just checks if action is `DestroyBeforeCreate() || CreateBeforeDestroy()`